retrieval/k8s/k8s_stats.go

```diff
@@ -124,6 +126,38 @@ func writeK8sResourceFile(workDir *os.File, resourceName string,
 	return err
 }
 
+//nolint:gocyclo
+func shouldSkipResource(k8Resource interface{}) bool {
+	// safe buffer to allow for longer lived resources to be ingested correctly
+	previousHour := time.Now().Add(-1 * time.Hour)
+	switch resource := k8Resource.(type) {
+	case *v1batch.Job:
+		if resource.Status.CompletionTime != nil &&
+			previousHour.After(resource.Status.CompletionTime.Time) {
+			return true
+		}
+		if resource.Status.Failed > 0 && resource.Status.StartTime != nil &&
+			previousHour.After(resource.Status.StartTime.Time) {
+			return true
+		}
+	case *corev1.Pod:
+		if resource.Status.Phase == corev1.PodSucceeded || resource.Status.Phase == corev1.PodFailed {
+			canSkip := true
+			for _, v := range resource.Status.ContainerStatuses {
+				if v.State.Terminated != nil && v.State.Terminated.FinishedAt.After(previousHour) {
+					canSkip = false
+				}
+			}
```

Comment:
Interest why do we need to this check as Succeeded/Failed are defined below
Succeeded: All containers in the Pod have terminated in success, and will not be restarted.
Failed: All containers in the Pod have terminated, and at least one container has terminated in failure. That is, the container either exited with non-zero status or was terminated by the system, and is not set for automatic restarting.

Reply:
My thought for keeping this data was that we may want information related to recently shut down pods.

We could remove all entries and not impact current allocation methodology.